### ISSUE TITLE
Very hard to debug

### ISSUE THREAD

**Issue by tomascharad** [mxstbr/react-boilerplate#541]
Hi,

I'm finding extremely difficult to debug the boilerplate.
When I set breakpoints with chrome's inspector it never stops.
Then only way to debug is to set a `debugger` statement or to `console.log`.
Also, the stacktrace is pretty weird.
Could you please help me? I would like to have an easier debugging experience.
Many thanks!

**Comment by mxstbr** [comment]
Where in the code are you setting those breakpoints?

**Comment by tomascharad** [comment]
On components for example.
May it be related to sourceMaps?
Take a look at this stackTrace.

<pre style='color:#000000;background:#ffffff;'>utils<span style='color:#008c00; '>.</span>js?<span style='color:#008c00; '>1</span>f12:<span style='color:#008c00; '>188</span> uncaught at getMovementPassport 
 ReferenceError: productGroupsInMovement is not defined
    at MovementPassport (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/<span style='color:#008c00; '>0.</span>chunk<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>42</span>:<span style='color:#008c00; '>61</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>38</span>:<span style='color:#008c00; '>3</span>)
    at ReactCompositeComponentWrapper<span style='color:#008c00; '>.</span>_constructComponentWithoutOwner (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/main<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>80</span>:<span style='color:#008c00; '>61</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>294</span>:<span style='color:#008c00; '>27</span>)
    at ReactCompositeComponentWrapper<span style='color:#008c00; '>.</span>_constructComponent (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/main<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>80</span>:<span style='color:#008c00; '>61</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>262</span>:<span style='color:#008c00; '>21</span>)
    at ReactCompositeComponentWrapper<span style='color:#008c00; '>.</span>mountComponent (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/main<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>80</span>:<span style='color:#008c00; '>61</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>181</span>:<span style='color:#008c00; '>21</span>)
    at Object<span style='color:#008c00; '>.</span>mountComponent (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/main<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>13</span>:<span style='color:#008c00; '>14113</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>46</span>:<span style='color:#008c00; '>35</span>)
    at ReactCompositeComponentWrapper<span style='color:#008c00; '>.</span>performInitialMount (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/main<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>80</span>:<span style='color:#008c00; '>61</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>357</span>:<span style='color:#008c00; '>34</span>)
    at ReactCompositeComponentWrapper<span style='color:#008c00; '>.</span>mountComponent (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/main<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>80</span>:<span style='color:#008c00; '>61</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>244</span>:<span style='color:#008c00; '>21</span>)
    at Object<span style='color:#008c00; '>.</span>mountComponent (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/main<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>13</span>:<span style='color:#008c00; '>14113</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>46</span>:<span style='color:#008c00; '>35</span>)
    at ReactCompositeComponentWrapper<span style='color:#008c00; '>.</span>_updateRenderedComponent (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/main<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>80</span>:<span style='color:#008c00; '>61</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>783</span>:<span style='color:#008c00; '>40</span>)
    at ReactCompositeComponentWrapper<span style='color:#008c00; '>.</span>_performComponentUpdate (eval at <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span> (http://localhost:<span style='color:#008c00; '>8080</span>/main<span style='color:#008c00; '>.</span>js:<span style='color:#008c00; '>80</span>:<span style='color:#008c00; '>61</span>), <span style='color:#a65700; '>&lt;</span><span style='color:#5f5035; '>anonymous</span><span style='color:#a65700; '>></span>:<span style='color:#008c00; '>753</span>:<span style='color:#008c00; '>10</span>)
</pre>

**Comment by mxstbr** [comment]
I'm pretty sure that's not something we can really change, that's mostly React just being the way it is I think…

**Comment by tomascharad** [comment]
I had to change the devTools to eval... not sure why cheap method wasn't working correctly. Will update this when I have more background.

**Comment by justingreenberg** [comment]
this behavior is expected, see https://github.com/mxstbr/react-boilerplate/pull/411

tldr; browser devtools require 1:1 line:col source mapping, obviously :)
